app/src/main/java/org/dhis2/uicomponents/map/TeiMarkers.kt

```diff
@@ -60,22 +59,39 @@ object TeiMarkers {
         canvas.drawBitmap(
             bitmap,
             canvas.width / 2f - bitmap.width / 2f,
-            canvas.height / 2f - bitmap.height / 2f,
+            canvas.height / 2f - bitmap.height / 2f - 5,
             null
         )
         return markerBitmap
     }
 
     fun getMarker(context: Context, teiImage: Drawable, color: Int): Bitmap {
-        val (canvas, markerBitmap) = initMarkerCanvas(context, color)
-
-        val paint = Paint()
-        paint.color = Color.WHITE
-        canvas.drawRect(15f, 10f, canvas.width - 15f, canvas.width - 25f, paint)
-
-        teiImage.setBounds(15, 10, canvas.width - 15, canvas.width - 25)
-        teiImage.draw(canvas)
+        return getMarker(context, drawableToBitmap(teiImage))
+    }
 
-        return markerBitmap
+    fun drawableToBitmap(drawable: Drawable): Bitmap {
+        if (drawable is BitmapDrawable) {
+            val bitmapDrawable: BitmapDrawable = drawable as BitmapDrawable
```

Comment:
you don't need to do the casting  drawable as BitmapDrawable , the compiler does it automatically once you check "If (drawable is BitmapDrawable)

Reply:
👍